AwesomeCalculator/app/src/main/java/com/example/drcreeper/awesomecalculator/historywriter/HistoryItemAdapter.java

```diff
@@ -0,0 +1,31 @@
+package com.example.drcreeper.awesomecalculator.historywriter;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.TextView;
+
+import com.example.drcreeper.awesomecalculator.R;
+
+import java.util.List;
+
+public class HistoryItemAdapter extends ArrayAdapter<String> {
+    public HistoryItemAdapter(Context context, List<String> list){
+        super(context, R.layout.history_item,list);
+    }
+
+    @NonNull
+    @Override
+    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
+        LayoutInflater inflater = LayoutInflater.from(getContext());
```

Comment:
Please add view recycling and ViewHolder pattern to optimize performance of adapter.